charts/datadog/templates/_kubernetes_apiserver_config.yaml

```
@@ -1,13 +1,19 @@
 {{- define "kubernetes_apiserver-config" -}}
-{{- if .Values.datadog.collectEvents -}}
+{{- if or (.Values.datadog.collectEvents) (.Values.clusterAgent.kubernetesApiserverCheck.disableUseComponentStatus) -}}
 kubernetes_apiserver.yaml: |-
   init_config:
   instances:
-    - filtering_enabled: {{ .Values.datadog.kubernetesEvents.filteringEnabled }}
+    - 
+{{- if .Values.datadog.collectEvents }}
+      filtering_enabled: {{ .Values.datadog.kubernetesEvents.filteringEnabled }}
       unbundle_events: {{ .Values.datadog.kubernetesEvents.unbundleEvents }}
       {{- if .Values.datadog.kubernetesEvents.unbundleEvents }}
       collected_event_types:
 {{ .Values.datadog.kubernetesEvents.collectedEventTypes | toYaml | nindent 8 }}
       {{- end -}}
+{{- end }}
+{{- if .Values.clusterAgent.kubernetesApiserverCheck.disableUseComponentStatus }}
+      use_component_status: false
+{{- end }}
 {{- end -}}
-{{- end -}}
+{{- end -}}
```

nit: Can you add an extra empty line at the bottom of the file?

```suggestion
{{- end -}}

```